src/main/java/com/arpnetworking/metrics/impl/TsdCompoundUnit.java

```diff
@@ -0,0 +1,356 @@
+ * which could result in an empty compound unit (e.g. Unit.NO_UNIT).
+ * Finally, two compound units are considered equal if their components
+ * are equal regardless of order (e.g. foot,pounds equals pound,feet).
+ *
+ * @author Ville Koskela (vkoskela at groupon dot com)
+ */
+public final class TsdCompoundUnit implements CompoundUnit {
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String getName() {
+        final StringBuilder stringBuilder = new StringBuilder();
+        final boolean numeratorParenthesis = _numeratorUnits.size() > 1 && !_denominatorUnits.isEmpty();
+        final boolean denominatorParenthesis = _denominatorUnits.size() > 1;
+        if (!_numeratorUnits.isEmpty()) {
+            if (numeratorParenthesis) {
+                stringBuilder.append("(");
+            }
+            for (final Unit unit : _numeratorUnits) {
+                stringBuilder.append(unit.getName());
+                stringBuilder.append("*");
+            }
+            stringBuilder.deleteCharAt(stringBuilder.length() - 1);
+            if (numeratorParenthesis) {
+                stringBuilder.append(")");
+            }
+        } else {
+            stringBuilder.append("1");
+        }
+        if (!_denominatorUnits.isEmpty()) {
+            stringBuilder.append("/");
+            if (denominatorParenthesis) {
+                stringBuilder.append("(");
+            }
+            for (final Unit unit : _denominatorUnits) {
+                stringBuilder.append(unit.getName());
+                stringBuilder.append("*");
+            }
+            stringBuilder.deleteCharAt(stringBuilder.length() - 1);
+            if (denominatorParenthesis) {
+                stringBuilder.append(")");
+            }
+        }
+        return stringBuilder.toString();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public List<Unit> getNumeratorUnits() {
+        return Collections.unmodifiableList(_numeratorUnits);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public List<Unit> getDenominatorUnits() {
+        return Collections.unmodifiableList(_denominatorUnits);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public boolean equals(final Object other) {
+        if (this == other) {
+            return true;
+        }
+
+        if (!(other instanceof TsdCompoundUnit)) {
+            return false;
+        }
+
+        final TsdCompoundUnit otherCompoundUnit = (TsdCompoundUnit) other;
+        return Objects.equals(_numeratorUnits, otherCompoundUnit._numeratorUnits)
+                && Objects.equals(_denominatorUnits, otherCompoundUnit._denominatorUnits);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public int hashCode() {
+        int hash = 17;
+        hash = hash * 31 + Objects.hashCode(_numeratorUnits);
+        hash = hash * 31 + Objects.hashCode(_denominatorUnits);
+        return hash;
```

Comment:
I'm not nuts about this since it creates an array every time the hash code is evaluated. However, I suppose there's no point in premature optimization. Changed.